articles/openstack/ostack-sso-rcfile.md

```diff
@@ -0,0 +1,54 @@
+---
+title: How to use the OpenStack API using an SSO enabled user | UKCloud Ltd
+description: Explains the changes needed to make use of the OpenStack API with an SSO enabled user.
+services: openstack
+author: Steve Relf
+
+toc_rootlink: How To
+toc_sub1: 
+toc_sub2:
+toc_sub3:
+toc_sub4:
+toc_title: OpenStack API with SSO users
+toc_fullpath: How To/ostack-sso-rcfile.md
+toc_mdlink: ostack-sso-rcfile.md
+---
+
+# How to use the OpenStack API using an SSO enabled user
+
+## Overview
+
+In order to use the OpenStack API with your SSO enabled user, you will need to make a few changes to your downloaded OpenStack RC file. This is to allow your API calls to be correctly passed to the SSO service.
+
+## Downloading your RC file.
+
+Please see [this guide](ostack-how-use-cli.md) section five for details on where to ger your RC file. You will require the v3 version. 
+
+## Modifying your RC file.
+
+Once you have downloaded your RC file you will need to add the following items to the bottom of the file.
+
+
+* export OS_IDENTITY_PROVIDER="sso"
+* export OS_PROTOCOL="oidc"
+* export OS_AUTH_TYPE="v3oidcpassword"
+
+You will also need to add the below items, and using the table below choose the correct entries for the OpenStack region you are connecting too.
+* export OS_CLIENT_ID=""
+* export OS_CLIENT_SECRET=""
+* export OS_ACCESS_TOKEN_ENDPOINT=""
+
+| Site | Variable | Value|
```

Comment:
I'd reconfigure this table slightly to make it a bit clearer:
Variable | Site | Value
---------|------|-------
`OS_CLIENT_ID` | Farnborough | cni.1.frn00006
&nbsp;              | Corsham       | cni.1.cor00005
`OS_CLIENT_SECRET` | Farnborough | User defined
&nbsp;                      | Corsham        | User defined
`OS_ACCESS_TOKEN_ENDPOINT` | Farnborough | https://idp.ukcloud.com/auth/realms/client-assured/protocol/openid-connect/token
&nbsp;                                          | Corsham       | https://idp.ukcloud.com/auth/realms/client-assured/protocol/openid-connect/token